room_assistance/indico_room_assistance/templates/assistance_request_occurrences.html

```diff
@@ -0,0 +1,67 @@
+{% extends 'forms/base_widget.html' %}
+
+{% from 'forms/_form.html' import form_field %}
+{% from '_switch.html' import switch %}
+
+{% block html %}
+    <div class="assistance-occurrences-widget i-form-field-fixed-width" id="{{ field.id }}-container">
+        <input type="hidden" name="{{ field.name }}" id="{{ field.id }}"
+               value="{{ field._value()|tojson|forceescape }}"
+               autofocus>
+        {%- for day in field.event.iter_days() -%}
+            <div class="occurrence">
+                <input type="text" class="datepicker" value="{{ day|format_date(format='code') }}"
+                       data-value="{{ day }}"
+                       disabled>
+                <input type="time" class="timepicker" placeholder="hh:mm" pattern="{{ time_regex_hhmm }}"
+                       value="{{ field._value()[day.isoformat()] or field.event_start_dt.time().strftime('%H:%M') }}"
+                       {{ 'disabled' if day.isoformat() not in field._value() }}
+                       required>
+                {{ switch(checked=(day.isoformat() in field._value())) }}
+            </div>
+        {%- endfor -%}
+    </div>
+{% endblock %}
+
+{% block javascript -%}
+    <script>
+        const $field = $('#{{ field.id }}');
+        const $occurrences = $('#{{ field.id }}-container .occurrence');
+
+        $occurrences
+            .find('.datepicker')
+            .datepicker({
+                dateFormat: 'dd/mm/yy',
+                disabled: true
+            })
+            .end()
```

Comment:
can't we just start a new statement here? I don't think jumping around in the selector stack makes code particularly readable

Reply:
We can, but at least to me using `end` makes this code a little bit more concise and readable.